### ISSUE TITLE
Broken Azure deployment for SAFE stack(net5.0,fable3)

### ISSUE THREAD

**Issue by Akash-Mair** [eiriktsarpalis/TypeShape#45]
### Description

Server fails to start for deployed webapp

```
Application: w3wp.exe
CoreCLR Version: 5.0.321.7212
.NET Version: 5.0.3
Description: The process was terminated due to an internal error in the .NET Runtime at IP 581BB105 (58000000) with exit code 80131506.
```
### Repro steps
In the root of the project

1. `dotnet tool restore` 
2. `dotnet run -- Azure`
3. Navigating to the deployed site you should see the 502 error listed below 

### Expected behavior

Once deployed I expect to see the basic SAFE template (to do app)

### Actual behavior

502 - Web server received an invalid response while acting as a gateway or proxy server.
There is a problem with the page you are looking for, and it cannot be displayed. When the Web server (while acting as a gateway or proxy) contacted the upstream content server, it received an invalid response from the content server.

### Known workarounds

Pinning the TypeShape version in the paket.dependencies to 8.0.1

### Related information 

Example Repo: https://github.com/CompositionalIT/broken-SAFEv3-deployment
Related Issue: https://github.com/Zaid-Ajaj/Fable.Remoting/issues/226

### EDIT - Full stack trace of error 

```
D:\home\site\wwwroot>dotnet Server.dll
Fatal error. Internal CLR error. (0x80131506)
 at System.RuntimeType.get_IsByRefLike()
 at TypeShape.Core.Core+TypeShapeImpl.resolveTypeShape(System.Type)
 at TypeShape.Core.Core+ShapeFSharpRecord`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]..ctor()
 at System.RuntimeMethodHandle.InvokeMethod(System.Object, System.Object[], System.Signature, Boolean, Boolean)
 at System.Reflection.RuntimeConstructorInfo.Invoke(System.Reflection.BindingFlags, System.Reflection.Binder, System.Object[], System.Globalization.CultureInfo)
 at TypeShape.Core.Core+TypeShapeImpl.activateGeneric(System.Type, System.Type[], System.Object[])
 at TypeShape.Core.Core.Activator.CreateInstanceGeneric.Static[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](Microsoft.FSharp.Core.FSharpOption`1<System.Type[]>, Microsoft.FSharp.Core.FSharpOption`1<System.Object[]>)
 at TypeShape.Core.Core+Shape.|FSharpRecord|_|(TypeShape)
 at Fable.Remoting.Server.Proxy.makeApiProxy[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](Fable.Remoting.Server.RemotingOptions`2<System.__Canon,System.__Canon>)
 at Fable.Remoting.Giraffe.GiraffeUtil.buildFromImplementation[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](Microsoft.FSharp.Core.FSharpFunc`2<Microsoft.AspNetCore.Http.HttpContext,System.__Canon>, Fable.Remoting.Server.RemotingOptions`2<Microsoft.AspNetCore.Http.HttpContext,System.__Canon>)
 at Fable.Remoting.Giraffe.Remoting.buildHttpHandler[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](Fable.Remoting.Server.RemotingOptions`2<Microsoft.AspNetCore.Http.HttpContext,System.__Canon>)
 at <StartupCode$Server>.$Server.main@()
```

**Comment by isaacabraham** [comment]
@Akash-Mair could you paste the full stack trace that you supplied me with as part of this issue? Thanks

**Comment by eiriktsarpalis** [comment]
Addressed by #48. See also https://www.nuget.org/packages/TypeShape/10.0.0-alpha1
